src/BlitHelper.cpp

```diff
@@ -390,4 +392,23 @@ namespace D3D12TranslationLayer
 
         m_pParent->PostRender(COMMAND_LIST_TYPE::GRAPHICS, e_GraphicsStateDirty);
     }
+
+    void BlitHelper::ResolveToNonMsaaIfNeeded( _Inout_ Resource **ppResource, _Inout_ std::vector<UINT> &newSubresourceIndices, const RECT& srcRect )
+    {
+        auto pResource = *ppResource;
+        if (pResource->AppDesc()->Samples() > 1)
+        {
+            assert( newSubresourceIndices.size() <= MAX_PLANES );
+
+            auto& cacheEntry = m_pParent->GetResourceCache().GetResource( pResource->AppDesc()->Format(), RectWidth( srcRect ), RectHeight( srcRect ) );

```

Comment:
See the comments above about ending up with a 3-pass approach, and how that can cause a read+write of the same resource in some cases:
```c++
                // If we're doing both of these, then our two-pass approach is going to turn into
                // a three-pass approach (first color convert to a cache entry, then swap RB channels
                // into another cache entry, then copy to the non-renderable destination).
                // In this case, passes one and two will both try to render to the cache entry,
                // with pass two trying to simultaneously read AND write from the cache entry.
                //
                // To prevent this, take ownership of the cache entry during pass one, so that pass two
                // allocates a new resource. The resource from pass one will then be destroyed rather than
                // being cached... but this seems okay since this should be *very* uncommon
```

We won't end up with a 2-pass color convert interacting here (since YUV resources can't be MSAA), but I think we could have an MSAA source, and a non-RTV dest, where we need to do:
1. Resolve to a same-size intermediate from the cache.
2. Use a shader to stretch to an RTV intermediate from the cache.
3. Copy to the final (non-RTV) destination.

In that case, steps 1 and 2 could end up using the same cached resource if we're not careful. I think we need another `TakeCacheEntryOwnership` call, or something similar here.

Reply:
Ah I see. I guess I wasn't running into that issue in the test app since the src and dest had different formats.